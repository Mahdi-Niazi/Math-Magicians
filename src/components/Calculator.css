* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.container-calc {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 400px;
  margin: 100px auto;
}

.input {
  width: 100%;
  height: 60px;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  background-color: #858694;
  color: #fff;
  font-size: 30px;
  padding: 17px;
  text-align: right;
  border: none;
  outline: none;
  box-shadow: 5px 5px 15px rgba(0, 0, 0, 0.2);
}

.parent {
  display: grid;
  grid-template-columns: auto auto auto auto;
  background-color: #e3dfdf;
  border-radius: 5px;
  box-shadow: 5px 5px 15px rgba(0, 0, 0, 0.2);
}

.button {
  background-color: rgba(199, 199, 199, 0.995);
  padding: 20px;
  font-size: 22px;
  text-align: center;
  margin: 1px;
  width: 98px;
  height: 80px;
  border: 0;
}

.button-zero {
  grid-column: 1/3;
  background-color: rgba(199, 199, 199, 0.995);
  border: 0;
}

.button-color {
  background-color: orange;
  padding: 20px;
  font-size: 15px;
  text-align: center;
  margin: 1px;
  width: 98px;
  height: 80px;
  border: 0;
}
