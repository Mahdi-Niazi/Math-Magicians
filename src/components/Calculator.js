import React, { useState } from 'react';
import calculate from '../logic/calculate';
import './Calculator.css';

const Calculator = () => {
  const [state, setState] = useState({
    total: null,
    next: null,
    operation: null,
  });
  const { total, operation, next } = state;
  const handleEvent = (e) => {
    const value = e.target.textContent;
    const calculat = calculate(state, value);
    setState(calculat);
  };
  return (
    <div className="container-calc">
      <div>
        <div className="input">
          {total}
          {operation}
          {next}
        </div>
        <div className="parent">
          <button type="button" className="button" onClick={handleEvent}>
            AC
          </button>
          <button type="button" className="button" onClick={handleEvent}>
            +/-
          </button>
          <button type="button" className="button" onClick={handleEvent}>
            %
          </button>
          <button type="button" className="button-color" onClick={handleEvent}>
            รท
          </button>
          <button type="button" className="button" onClick={handleEvent}>
            7
          </button>
          <button type="button" className="button" onClick={handleEvent}>
            8
          </button>
          <button type="button" className="button" onClick={handleEvent}>
            9
          </button>
          <button type="button" className="button-color" onClick={handleEvent}>
            x
          </button>
          <button type="button" className="button" onClick={handleEvent}>
            4
          </button>
          <button type="button" className="button" onClick={handleEvent}>
            5
          </button>
          <button type="button" className="button" onClick={handleEvent}>
            6
          </button>
          <button type="button" className="button-color" onClick={handleEvent}>
            -
          </button>
          <button type="button" className="button" onClick={handleEvent}>
            1
          </button>
          <button type="button" className="button" onClick={handleEvent}>
            2
          </button>
          <button type="button" className="button" onClick={handleEvent}>
            3
          </button>
          <button type="button" className="button-color" onClick={handleEvent}>
            +
          </button>
          <button type="button" className="button-zero" onClick={handleEvent}>
            0
          </button>
          <button type="button" className="button" onClick={handleEvent}>
            .
          </button>
          <button type="button" className="button-color" onClick={handleEvent}>
            =
          </button>
        </div>
      </div>
    </div>
  );
};

export default Calculator;
