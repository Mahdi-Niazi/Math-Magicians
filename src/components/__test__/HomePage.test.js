import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Calculator from '../Calculator';

describe('Calculator Page', () => {
  it('Should', () => {
    render(<Calculator />);
    const h1 = screen.getByTestId('h2Id');
    expect(h1.innerHTML).toBe("Let's do some maths");
  });

  it('Render the Calculator Page user interaction', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
