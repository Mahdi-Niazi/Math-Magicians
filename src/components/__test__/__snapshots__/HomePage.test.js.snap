// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Page Render the Calculator Page user interaction 1`] = `
<div
  className="container-calc"
>
  <h1
    data-testid="h2Id"
  >
    Let's do some maths
  </h1>
  <div>
    <div
      className="input"
    />
    <div
      className="parent"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="button-color"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="button-color"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="button-color"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="button-color"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="button-zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="button-color"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
